test/controlplane/pod/hostport/hostport.go

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package hostport
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	operatorOption "github.com/cilium/cilium/operator/option"
+	agentOption "github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/test/controlplane"
+	"github.com/cilium/cilium/test/controlplane/services/helpers"
+	"github.com/cilium/cilium/test/controlplane/suite"
+)
+
+func init() {
+	suite.AddTestCase("Pod/HostPort", testHostPort)
+
+}
+
+func testHostPort(t *testing.T) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	abs := func(f string) string { return path.Join(cwd, "pod", "hostport", f) }
+
+	modConfig := func(daemonCfg *agentOption.DaemonConfig, _ *operatorOption.OperatorConfig) {}
+
+	k8sVersions := controlplane.K8sVersions()
+	// We only need to test the last k8s version
+	test := suite.NewControlPlaneTest(t, "pod-hostport-control-plane", k8sVersions[len(k8sVersions)-1])
+	defer test.StopAgent()
+
+	// Feed in initial state and start the agent.
+	test.
+		UpdateObjectsFromFile(abs("init.yaml")).
+		SetupEnvironment(modConfig).
+		StartAgent().
```

Comment:
@joamaki we are stuck with the test, it hangs at this point and it never move forward, we compared with the other tests and we can't see anything obvious, can you help us to understand what is missing?

It seems that it is waiting for a local node initialization
```
oroutine 118 [chan receive, 9 minutes]:
github.com/cilium/cilium/daemon/cmd.(*localNodeInitializer).getK8sLocalNode(0xc000e9c6f0, {0x3ed7ef8?, 0xc000527b80?})
        /usr/local/google/home/aojea/src/cilium/daemon/cmd/local_node_init.go:87 +0xec
github.com/cilium/cilium/daemon/cmd.(*localNodeInitializer).initFromK8s(0xc000e9c6f0, {0x3ed7ef8?, 0xc000527b80?}, 0xc0001ea178)
        /usr/local/google/home/aojea/src/cilium/daemon/cmd/local_node_init.go:101 +0x5b
github.com/cilium/cilium/daemon/cmd.(*localNodeInitializer).InitLocalNode(0x4b1886?, {0x3ed7ef8, 0xc000527b80}, 0xc0001ea178)
        /usr/local/google/home/aojea/src/cilium/daemon/cmd/local_node_init.go:48 +0x9e
github.com/cilium/cilium/pkg/node.NewLocalNodeStore.func1({0x7f30000e7d00?, 0xc000527b80})
        /usr/local/google/home/aojea/src/cilium/pkg/node/local_node_store.go:81 +0x187
github.com/cilium/cilium/pkg/hive.Hook.Start(...)
        /usr/local/google/home/aojea/src/cilium/pkg/hive/lifecycle.go:35
github.com/cilium/cilium/pkg/hive.(*DefaultLifecycle).Start(0xc00123e420, {0x3ed7f30?, 0xc000226008?})
        /usr/local/google/home/aojea/src/cilium/pkg/hive/lifecycle.go:104 +0x2ee
github.com/cilium/cilium/pkg/hive.(*Hive).Start(0xc00127cb40, {0x3ed7f30, 0xc000226008})
        /usr/local/google/home/aojea/src/cilium/pkg/hive/hive.go:295 +0x105
github.com/cilium/cilium/test/controlplane/suite.startCiliumAgent(0xc000303ba0, {0x3f17488?, 0xc0012291c0}, {0x3ec03d0?, 0x63b9620})
        /usr/local/google/home/aojea/src/cilium/test/controlplane/suite/agent.go:105 +0x785
github.com/cilium/cilium/test/controlplane/suite.(*ControlPlaneTest).StartAgent(0xc001261770, {0x0, 0x0, 0x0})
        /usr/local/google/home/aojea/src/cilium/test/controlplane/suite/testcase.go:141 +0xe5
github.com/cilium/cilium/test/controlplane/services/hostport.testHostPort(0xc000303ba0)
        /usr/local/google/home/aojea/src/cilium/test/controlplane/services/hostport/hostport.go:42 +0x1f2
testing.tRunner(0xc000303ba0, 0x3ae3388)
        /usr/local/google/home/aojea/sdk/go1.20.4/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
        /usr/local/google/home/aojea/sdk/go1.20.4/src/testing/testing.go:1629 +0x3ea
```

Reply:
found the problem https://github.com/cilium/cilium/pull/22587/files#r1244016712